pandas/plotting/_matplotlib/style.py

```diff
@@ -57,28 +57,8 @@ def random_color(column):
             raise ValueError("color_type must be either 'default' or 'random'")
 
     if isinstance(colors, str):
-        conv = matplotlib.colors.ColorConverter()
-
-        def _maybe_valid_colors(colors):
-            try:
-                [conv.to_rgba(c) for c in colors]
-                return True
-            except ValueError:
-                return False
-
-        # check whether the string can be convertible to single color
-        maybe_single_color = _maybe_valid_colors([colors])
-        # check whether each character can be convertible to colors
-        maybe_color_cycle = _maybe_valid_colors(list(colors))
```

Comment:
are these not relevant / texted? (e.g. the axes.prop_cycle) or is this handled by conv.to_rbga now?

Reply:
I guess that this was the way of handling compatibility with matplotlib <= 2.0.
If the color 'CN' is passed, then it would pick N-th color from "axes.prop_cycle".
Before matplotlib 2.0 there were no CN colors, so this is probably why this construction is here.